PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp

```diff
@@ -703,29 +703,33 @@ vector<wstring> Finder::getResultFilePaths(bool onlyInSelectedText) const
 		toLine = _scintView.execute(SCI_GETLINECOUNT) - 1;
 	}
 
+	size_t len = _pMainFoundInfos->size();  // First, get the number of elements in the container
 	for (size_t line = fromLine; line <= toLine; ++line)
 	{
+		bool found = false;  // Was it found?
 		const int lineFoldLevel = _scintView.execute(SCI_GETFOLDLEVEL, line) & SC_FOLDLEVELNUMBERMASK;
 		if (lineFoldLevel == fileHeaderLevel)
 		{
-			wstring lineStr = _scintView.getLine(line);
-
-			// fileHeaderLevel line format examples:
-			// spacespaceD:\folder\file.ext (2 hits)
-			// spacespacenew 1 (1 hit)
-			const size_t startIndex = 2;  // for number of leading spaces
-			auto endIndex = lineStr.find_last_of(L'(');
-			--endIndex;  // adjust for space in front of (
-			wstring path = lineStr.substr(startIndex, endIndex - startIndex);
-
-			// make sure that path is not already in before adding
-			if (std::find(paths.begin(), paths.end(), path) == paths.end())
+			line++;  // Move to the next line
+			if (line < len)
+				found = true;  // Found it
+		}
+		else if (lineFoldLevel == resultLevel)
+		{
+			if (line < len)
+				found = true;  // Found it
+		}
+		if (found)
+		{
+			wstring& path = (*_pMainFoundInfos)[line]._fullPath;  // Get the path from the container
+			if (path.find('\\') != std::wstring::npos && std::find(paths.begin(), paths.end(), path) == paths.end())  // Contains a path separator and does not exist in the container

```

Comment:
IMO this line's logic introduces a new bug; should have read https://github.com/notepad-plus-plus/notepad-plus-plus/issues/15741#issuecomment-2480863202 and I have elaborated newly upon it https://github.com/notepad-plus-plus/notepad-plus-plus/issues/15741#issuecomment-2481196248